### DISCUSSION TITLE
With yarn 3.2.1, webpack compile shows 16 warnings as below, can't you make them invisible?

### DISCUSSION THREAD

**Discussion by seleondevel** [kimai/kimai#5013] [Plugins]
Hello!
With yarn3.2.1, webpack compile shows 16 warnings as below, can't you make them invisible?
With 16 alerts popping up, taking too long to compile, do I need to downgrade the Sass version or modify the Sass file myself?
Or, is there a way to ignore warnings in webpack settings?

yarn watch
Running webpack ...

 WARN  Compiled with 16 warnings                                                                                                                                                  5:21:13 PM

 warn  in ./assets/sass/_app.scss                                                                                                                                                 5:21:13 PM

Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 25, column 2 of file:///kimai/node_modules/bootstrap/scss/forms/_form-select.scss:25:2:
Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

25 |   appearance: none;

node_modules/bootstrap/scss/forms/_form-select.scss 26:3           @import
node_modules/bootstrap/scss/_forms.scss 4:9                        @import
node_modules/@tabler/core/src/scss/_bootstrap-components.scss 9:9  @import
node_modules/@tabler/core/src/scss/_core.scss 2:9                  @import
node_modules/@tabler/core/src/scss/tabler.scss 1:9                 @import
assets/sass/_app.scss 10:9                                         root stylesheet

 warn  in ./assets/sass/_app.scss                                                                                                                                                 5:21:13 PM

Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 35, column 4 of file:///kimai/node_modules/bootstrap/scss/forms/_form-range.scss:35:4:
Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

35 |     appearance: none;

node_modules/bootstrap/scss/forms/_form-range.scss 36:5            @import
node_modules/bootstrap/scss/_forms.scss 6:9                        @import
node_modules/@tabler/core/src/scss/_bootstrap-components.scss 9:9  @import
node_modules/@tabler/core/src/scss/_core.scss 2:9                  @import
node_modules/@tabler/core/src/scss/tabler.scss 1:9                 @import
assets/sass/_app.scss 10:9                                         root stylesheet

 warn  in ./assets/sass/_app.scss                                                                                                                                                 5:21:13 PM

Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 53, column 4 of file:////kimai/node_modules/@tabler/core/src/scss/layout/_root.scss:53:4:
Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

53 |     --#{$prefix}font-weight-#{$name}: #{$value};

node_modules/@tabler/core/src/scss/layout/_root.scss 54:5  @import
node_modules/@tabler/core/src/scss/_core.scss 6:9          @import
node_modules/@tabler/core/src/scss/tabler.scss 1:9         @import
assets/sass/_app.scss 10:9                                 root stylesheet

 warn  in ./assets/sass/_app.scss                                                                                                                                                 5:21:13 PM

Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 56, column 13 of file:////kimai/node_modules/bootstrap/scss/vendor/_rfs.scss:56:13:
Passing percentage units to the global abs() function is deprecated.
In the future, this will emit a CSS abs() function to be resolved by the browser.
To preserve current behavior: math.abs(85.714285%)
To emit a CSS abs() now: abs(#{85.714285%})
More info: https://sass-lang.com/d/abs-percent

56 |   $dividend: abs($dividend);

node_modules/bootstrap/scss/vendor/_rfs.scss 57:14       divide()
node_modules/@tabler/core/src/scss/ui/_badges.scss 8:41  @import
node_modules/@tabler/core/src/scss/_core.scss 17:9       @import
node_modules/@tabler/core/src/scss/tabler.scss 1:9       @import
assets/sass/_app.scss 10:9                               root stylesheet

 warn  in ./assets/sass/_app.scss                                                                                                                                                 5:21:13 PM

Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 56, column 13 of file:////kimai/node_modules/bootstrap/scss/vendor/_rfs.scss:56:13:
Passing percentage units to the global abs() function is deprecated.
In the future, this will emit a CSS abs() function to be resolved by the browser.
To preserve current behavior: math.abs(100%)
To emit a CSS abs() now: abs(#{100%})
More info: https://sass-lang.com/d/abs-percent

56 |   $dividend: abs($dividend);

node_modules/bootstrap/scss/vendor/_rfs.scss 57:14                 divide()
node_modules/bootstrap/scss/mixins/_grid.scss 59:12                row-cols()
node_modules/bootstrap/scss/mixins/_grid.scss 85:13                @content
node_modules/bootstrap/scss/mixins/_breakpoints.scss 68:5          media-breakpoint-up()
node_modules/bootstrap/scss/mixins/_grid.scss 72:5                 make-grid-columns()
node_modules/bootstrap/scss/_grid.scss 38:3                        @import
node_modules/@tabler/core/src/scss/_bootstrap-components.scss 7:9  @import
node_modules/@tabler/core/src/scss/_core.scss 2:9                  @import
node_modules/@tabler/core/src/scss/tabler.scss 1:9                 @import
assets/sass/_app.scss 10:9                                         root stylesheet

 warn  in ./assets/sass/_app.scss                                                                                                                                                 5:21:13 PM

Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 57, column 12 of file:////kimai/node_modules/bootstrap/scss/vendor/_rfs.scss:57:12:
Passing percentage units to the global abs() function is deprecated.
In the future, this will emit a CSS abs() function to be resolved by the browser.
To preserve current behavior: math.abs(100%)
To emit a CSS abs() now: abs(#{100%})
More info: https://sass-lang.com/d/abs-percent

57 |   $divisor: abs($divisor);

node_modules/bootstrap/scss/vendor/_rfs.scss 58:13       divide()
node_modules/@tabler/core/src/scss/ui/_badges.scss 8:41  @import
node_modules/@tabler/core/src/scss/_core.scss 17:9       @import
node_modules/@tabler/core/src/scss/tabler.scss 1:9       @import
assets/sass/_app.scss 10:9                               root stylesheet

 warn  in ./assets/sass/_app.scss                                                                                                                                                 5:21:13 PM

Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 61, column 4 of file:////kimai/node_modules/bootstrap/scss/forms/_form-range.scss:61:4:
Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

61 |     appearance: none;

node_modules/bootstrap/scss/forms/_form-range.scss 62:5            @import
node_modules/bootstrap/scss/_forms.scss 6:9                        @import
node_modules/@tabler/core/src/scss/_bootstrap-components.scss 9:9  @import
node_modules/@tabler/core/src/scss/_core.scss 2:9                  @import
node_modules/@tabler/core/src/scss/tabler.scss 1:9                 @import
assets/sass/_app.scss 10:9                                         root stylesheet

 warn  in ./assets/sass/_app.scss                                                                                                                                                 5:21:13 PM

Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 62, column 4 of file:////kimai/node_modules/bootstrap/scss/_modal.scss:62:4:
Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

62 |     transform: $modal-fade-transform;

node_modules/bootstrap/scss/_modal.scss 63:5                        @import
node_modules/@tabler/core/src/scss/_bootstrap-components.scss 26:9  @import
node_modules/@tabler/core/src/scss/_core.scss 2:9                   @import
node_modules/@tabler/core/src/scss/tabler.scss 1:9                  @import
assets/sass/_app.scss 10:9                                          root stylesheet

 warn  in ./assets/sass/_app.scss                                                                                                                                                 5:21:13 PM

Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
49 repetitive deprecation warnings omitted.

 warn  in ./assets/sass/_invoice.scss                                                                                                                                             5:21:13 PM

Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 37, column 4 of file:////kimai/node_modules/bootstrap/scss/_type.scss:37:4:
Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

37 |     font-family: $display-font-family;

node_modules/bootstrap/scss/_type.scss 38:5  @import
assets/sass/_invoice.scss 16:9               root stylesheet

 warn  in ./assets/sass/_invoice.scss                                                                                                                                             5:21:13 PM

Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 38, column 4 of file:////kimai/node_modules/bootstrap/scss/_type.scss:38:4:
Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

38 |     font-style: $display-font-style;

node_modules/bootstrap/scss/_type.scss 39:5  @import
assets/sass/_invoice.scss 16:9               root stylesheet

 warn  in ./assets/sass/_invoice.scss                                                                                                                                             5:21:13 PM

Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 39, column 4 of file:////kimai/node_modules/bootstrap/scss/_type.scss:39:4:
Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

39 |     font-weight: $display-font-weight;

node_modules/bootstrap/scss/_type.scss 40:5  @import
assets/sass/_invoice.scss 16:9               root stylesheet

 warn  in ./assets/sass/_invoice.scss                                                                                                                                             5:21:13 PM

Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 56, column 13 of file:////kimai/node_modules/bootstrap/scss/vendor/_rfs.scss:56:13:
Passing percentage units to the global abs() function is deprecated.
In the future, this will emit a CSS abs() function to be resolved by the browser.
To preserve current behavior: math.abs(100%)
To emit a CSS abs() now: abs(#{100%})
More info: https://sass-lang.com/d/abs-percent

56 |   $dividend: abs($dividend);

node_modules/bootstrap/scss/vendor/_rfs.scss 57:14         divide()
node_modules/bootstrap/scss/mixins/_grid.scss 59:12        row-cols()
node_modules/bootstrap/scss/mixins/_grid.scss 85:13        @content
node_modules/bootstrap/scss/mixins/_breakpoints.scss 68:5  media-breakpoint-up()
node_modules/bootstrap/scss/mixins/_grid.scss 72:5         make-grid-columns()
node_modules/bootstrap/scss/_grid.scss 38:3                @import
assets/sass/_invoice.scss 17:9                             root stylesheet

 warn  in ./assets/sass/_invoice.scss                                                                                                                                             5:21:13 PM

Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 501, column 2 of file:////kimai/node_modules/bootstrap/scss/_reboot.scss:501:2:
Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

501 |   font-weight: $legend-font-weight;

node_modules/bootstrap/scss/_reboot.scss 502:3  @import
assets/sass/_invoice.scss 15:9                  root stylesheet

 warn  in ./assets/sass/_invoice.scss                                                                                                                                             5:21:13 PM

Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
Deprecation Warning on line 502, column 2 of file:////kimai/node_modules/bootstrap/scss/_reboot.scss:502:2:
Sass's behavior for declarations that appear after nested
rules will be changing to match the behavior specified by CSS in an upcoming
version. To keep the existing behavior, move the declaration above the nested
rule. To opt into the new behavior, wrap the declaration in `& {}`.

More info: https://sass-lang.com/d/mixed-decls

502 |   line-height: inherit;

node_modules/bootstrap/scss/_reboot.scss 503:3  @import
assets/sass/_invoice.scss 15:9                  root stylesheet

 warn  in ./assets/sass/_invoice.scss                                                                                                                                             5:21:13 PM

Module Warning (from ./node_modules/sass-loader/dist/cjs.js):
21 repetitive deprecation warnings omitted.

webpack compiled with 16 warnings

**Comment by seleondevel** [comment]
Is there a setting in kimai/webpack.config.js that ignores warnings?

**Reply by seleondevel** [reply]
Is there a way to compile webpack efficiently in Kimai?

**Reply by kevinpapst** [reply]
So you think spamming is a good way to ask question? Unsubscribed.

**Comment by kevinpapst** [comment]
Check where the errors come from. Now go to Webpack, Node, Bootstrap or Tabler and ask them.
